synapse/lib/crypto/passwd.py

```diff
@@ -104,3 +106,59 @@ async def checkShadowV2(passwd: AnyStr, shadow: Dict) -> bool:
     if func is None:
         raise s_exc.CryptoErr(mesg=f'type [{ptyp}] does not map to a known function', valu=ptyp)
     return await func(passwd=passwd, shadow=shadow)
+
+async def generateApiKey(iden=None):
+    if iden is None:
+        iden = s_common.guid()
+    else:
+        if not s_common.isguid(iden):
+            raise s_exc.CryptoErr(mesg=f'Invalid iden provided: {iden}, must be a guid.')
+
+    iden_buf = s_common.uhex(iden)
+
+    secv = s_common.guid()
+    secv_buf = s_common.uhex(secv)
+    valu = b''
+
+    # Mix the iden and secret together. This produces an avalanche effect where
```

Comment:
lets have a quick goback about key format :+1:

Reply:
Discussed this.